### PR TITLE
Update to kotlin 1.8.0 and gradle 7.6

### PR DESCRIPTION
**Pull request by NoComment1105**
The binary compatibility validator plugin had to be moved into the main buildscript due to a quirk with kotlin file meta data. I have ensured Apis won't be built in places they aren't supposed too.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -89,7 +90,7 @@ allprojects {
         withType<KotlinCompile> {
             kotlinOptions {
                 jvmTarget = javaVersion.toString()
-                languageVersion = rootProject.libs.plugins.kotlin.get().version.strictVersion
+                //languageVersion = rootProject.libs.plugins.kotlin.get().version.strictVersion
```
Comment (SilverAndro) [inline build.gradle.kts:93]: Why is this commented out? Should have a TODO to add it back probably

**Review by Oliver-makes-code — APPROVED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (NoComment1105) [reply build.gradle.kts]: Added it back. Required a little extra parsing of the string, presumable due to changes in how gradle catalogs are handled. All works as normal though

File: build.gradle.kts [hunk 2]
```diff
@@ -89,7 +90,8 @@ allprojects {
         withType<KotlinCompile> {
             kotlinOptions {
                 jvmTarget = javaVersion.toString()
-                languageVersion = rootProject.libs.plugins.kotlin.get().version.strictVersion
+                languageVersion =
+                    rootProject.libs.plugins.kotlin.get().version.requiredVersion.split(".").dropLast(1).joinToString(".")
```
Comment (sschr15) [inline build.gradle.kts:94]:
```suggestion
                    rootProject.libs.plugins.kotlin.get().version.requiredVersion.substringBeforeLast(".")
```

**Review by sschr15 — APPROVED** [review]
Just the one switch to a cleaner stdlib function

**Review by PotatoPresident — APPROVED** [review]

**Review by SilverAndro — APPROVED** [review]

**Pull request merged**